.github/workflows/jobs.yml

```diff
@@ -221,30 +221,31 @@ jobs:
           make check
   windows_latest_cmake:
     runs-on: windows-latest
+    defaults:
+      run:
+        # Use MSYS2 as default shell
+        shell: msys2 {0}
     steps:
-      - name: Clone and check out repository code
```

Comment:
Can you set back this whole step? We had the CI broken (c45094c7583485b18d298423a6f5e66a130d3af4) because the branch checkout-ed by the CI job was not the one containing the modified code...

Reply:
I don't think that is necessary. Especially, `fetch-depth: 0` is overkill if you don't switch between branches or revisions anyway in any later build step.

Reply:
> The "Install MSYS2 build environment" step should be inserted after the "Clone and check out repository code" step.

We need `git` for that step. So, it should be after MSYS2 is installed.

Reply:
You might be under the impression that you need to select these specific `repository` and `ref` here. But that is not true.

When you selected the trigger `pull_request_target`, the action checked out the PR *base* branch. But the trigger for this action is `pull_request`. That selects the *merge* branch and commit.
See: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

Reply:
Totally believe you when you say it may be overkill or not necessary.

But, I prefer to have all jobs shaped the same way: easier to maintain, and, when needed changes are applied to all jobs at once. So for now, I prefer all jobs to be consistent (as it does not hurt)